lib/std/io/multi_out_stream.zig

```diff
@@ -0,0 +1,54 @@
+const std = @import("../std.zig");
+const io = std.io;
+const testing = std.testing;
+
+/// Takes a tuple of streams, and constructs a new stream that writes to all of them
+pub fn MultiOutStream(comptime OutStreams: type) type {
+    comptime var ErrSet = error{};
+    inline for (@typeInfo(OutStreams).Struct.fields) |field| {
+        const StreamType = field.field_type;
+        ErrSet = ErrSet || StreamType.Error;
+    }
+
+    return struct {
+        const Self = @This();
+
+        streams: OutStreams,
+
+        pub const Error = ErrSet;
+        pub const OutStream = io.OutStream(*Self, Error, write);
+        pub fn outStream(self: *Self) OutStream {
+            return .{ .context = self };
+        }
+
+        pub fn write(self: *Self, bytes: []const u8) Error!usize {
+            if (comptime self.streams.len == 0) return bytes.len;
+
+            // only first stream is allowed to do a partial write
+            // all subsequent streams do a `.writeAll`
+            const bytes_to_write = try self.streams[0].write(bytes);
+            const slice_to_write = bytes[0..bytes_to_write];
+            comptime var i = 1;
+            inline while (i < self.streams.len) : (i += 1) {
+                const stream = self.streams[i];
+                try stream.writeAll(slice_to_write);
+            }
+            return bytes_to_write;
```

Comment:
Maybe summarize all bytes written?

Reply:
no.... that's not what the return value of an out stream's `write` means.

Reply:
Why no?
Suppose you wrote 1 byte into stream1 and 1 byte into stream2. Obviously, you wrote 2 bytes.
E.g. you'll get the wrong result for `CountingOutStream` with `MultiOutStream`.

Reply:
The purpose of the return value is to know how many bytes of the input argument were written out, and can hence be discarded.